packages/opencv-python/meta.yaml

```diff
@@ -0,0 +1,144 @@
+package:
+  name: opencv-python
+  version: 4.5.5.64
+about:
+  home: https://github.com/skvark/opencv-python
+  PyPI: https://pypi.org/project/opencv-python
+  summary: Wrapper package for OpenCV python bindings.
+  license: MIT
+source:
+  url: https://files.pythonhosted.org/packages/3c/61/ee4496192ed27f657532fdf0d814b05b9787e7fc5122ed3ca57282bae69c/opencv-python-4.5.5.64.tar.gz
+  sha256: f65de0446a330c3b773cd04ba10345d8ce1b15dcac3f49770204e37602d0b3f7
+  extras:
+    - [cmake/OpenCVFindLibsGrfmt.cmake, opencv/cmake/OpenCVFindLibsGrfmt.cmake]
+    - [
+        cmake/detect_ffmpeg.cmake,
+        opencv/modules/videoio/cmake/detect_ffmpeg.cmake,
+      ]
+  patches:
+    - patches/0001-Enable-file-system.patch
+
+requirements:
+  run:
+    - numpy
+    - ffmpeg
+build:
+  cxxflags: |
+    -fPIC
+    -s USE_ZLIB=1
+    -s USE_LIBJPEG=1
+    -s USE_LIBPNG=1
+    -s SIDE_MODULE=1
+  ldflags: |
+    -ljpeg
+    -lz
+    -lpng
+
+  # Note on CMAKE_ARGS:
+  #   CMake args are adopted from OpenCV.js (https://github.com/opencv/opencv/blob/4.x/platforms/js/build_js.py)
+  #   But we support more of modules than OpenCV.js.
+  #
+  # Note on CMAKE_TOOLCHAIN_FILE:
+  #   We don't want to use toolchain file provided by Emscripten,
+  #   because our build script hijack gcc, c++, ... and replace it with emcc, em++, ..., instead of calling them directly.
+  #
+  # List of OpenCV modules can be found at: https://docs.opencv.org/4.x/
+  # Build configs can be found at: https://docs.opencv.org/4.x/db/d05/tutorial_config_reference.html
+
+  script: |
+    pip install scikit-build
+
+    export VERBOSE=1
+    export NUMPY_INCLUDE_DIR="$HOSTINSTALLDIR/lib/python$PYMAJOR.$PYMINOR/site-packages/numpy/core/include/"
+    export EMSCRIPTEN="$PYODIDE_ROOT/emsdk/emsdk/upstream/emscripten/"
+    export FFMPEG_ROOT="$PYODIDE_ROOT/packages/ffmpeg/build/ffmpeg-4.4.1/lib"
+
+    export CMAKE_ARGS=" \
+    -DCMAKE_TOOLCHAIN_FILE=$PYODIDE_ROOT/packages/opencv-python/cmake/Config.cmake \
+    -DPYTHON3_INCLUDE_PATH=$PYTHONINCLUDE \
+    -DPYTHON3_LIBRARY=$PYTHONINCLUDE/../libpython$PYMAJOR.$PYMINOR.a \
+    -DPYTHON3_VERSION_MAJOR=$PYMAJOR \
+    -DPYTHON3_VERSION_MINOR=$PYMINOR \
+    -DPYTHON3_NUMPY_INCLUDE_DIRS=$NUMPY_INCLUDE_DIR \
+    \
+    -DWITH_ADE=ON \
+    -DWITH_JPEG=ON \
+    -DWITH_PNG=ON \
+    -DWITH_WEBP=ON \
+    \
+    -DBUILD_opencv_python3=ON \
+    -DBUILD_opencv_world=ON \
+    -DBUILD_opencv_imgcodecs=ON \
+    -DBUILD_opencv_videoio=ON \
+    -DBUILD_opencv_gapi=ON \
+    -DBUILD_opencv_photo=ON \
+    -DBUILD_opencv_stitching=ON \
+    -DBUILD_opencv_highgui=ON \
+    -DBUILD_opencv_features2d=ON \
+    -DBUILD_opencv_flann=ON \
+    -DBUILD_opencv_calib3d=ON \
+    -DBUILD_opencv_dnn=ON \
+    -DBUILD_opencv_ml=ON \
+    -DBUILD_opencv_objdetect=ON \
+    -DWITH_OPENCL=OFF \
+    -DOPENCV_DNN_OPENCL=OFF \
+    -DWITH_PROTOBUF=ON \
```

Comment:
The protobuf library is currently built with opencv. As it is used in many deep learning packages, I'll try to make it a standalone package in a seperate PR (for now, opencv is the only one which uses protobuf).

Reply:
I was able to build libprotobuf in a separate package, but linking it with opencv generates a mysterious conflict between protobuf and ffmpeg. It looks like opencv builds [not all modules of protobuf](https://github.com/opencv/opencv/blob/66f3c2673cd424a9c282fb0c8622a4ae43d3e81d/3rdparty/protobuf/CMakeLists.txt#L56), so I'll let it build it's custom protobuf for now.